common/thorhelper/thorsoapcall.cpp

```diff
@@ -552,23 +557,34 @@ class BlackLister : public CInterface, implements IThreadFactory
 } *blacklist;
 
 static IPersistentHandler* persistentHandler = nullptr;
-static CriticalSection persistentCrit;
-static std::atomic<bool> persistentInitDone{false};
+static CriticalSection globalFeatureCrit;
+static std::atomic<bool> globalFeaturesInitDone{false};
+static std::atomic<bool> mapUrlsToSecrets{false};
+static std::atomic<bool> warnIfUrlNotMappedToSecret{false};
 
-void initPersistentHandler()
+void initGlobalFeatures()
 {
-    CriticalBlock block(persistentCrit);
-    if (!persistentInitDone)
+    CriticalBlock block(globalFeatureCrit);
+    if (!globalFeaturesInitDone)
     {
-#ifndef _CONTAINERIZED
-        int maxPersistentRequests = queryEnvironmentConf().getPropInt("maxHttpCallPersistentRequests", 0);
-#else
-        Owned<IPropertyTree> conf = getComponentConfig();
-        int maxPersistentRequests = conf->getPropInt("@maxHttpCallPersistentRequests", 0);
-#endif
+        int maxPersistentRequests = 0;
+        if (isContainerized())
```

Comment:
@ghalliday I'm not sure how best to make that work.  But if we were going to do it for bare-metal now is the time otherwise the way it's configured would change over time.
The process this is running in wouldn't be esp, it would be roxie, or eclagent, or thor.  I don't think when those are running on bare-metal they have a common way of getting at there config?  I could be wrong but in a simple test I didn't see the componentConfig getting set.

Reply:
Ok much better.  It's now using the component config in both cloud and bare-metal.